Content.Client/Chat/ChatManager.cs

```diff
@@ -104,5 +145,81 @@ private void _onChatBoxTextSubmitted(ChatBox chatBox, string text)
                 }
             }
         }
+
+        private void _onFilterButtonToggled(ChatBox chatBox, Button.ButtonEventArgs e)
+        {
+            switch (e.Button.Name)
+            {
+                case "OOC":
+
+                _OOCstate = !_OOCstate;
+                if (_OOCstate)
+                {
+                    filteredChannels.Add(ChatChannel.OOC);
+                    break;
+
+                } else
+                {
+                    filteredChannels.Remove(ChatChannel.OOC);
+                    _currentChatBox.contents.Clear();
+                    RepopulateChat(filteredHistory);
+
+                    break;
+                }
+
+                default:
+
+                _ALLstate = !_ALLstate;
+                if (_ALLstate)
+                {
+                    foreach (string enumString in ChatChannel.GetNames(typeof(ChatChannel)))
+                    {
+                        ChatChannel.TryParse(enumString, out ChatChannel channel);
+                        filteredChannels.Add(channel);
+                    }
+                }
+                else 
+                {
+                    foreach (string enumString in ChatChannel.GetNames(typeof(ChatChannel)))
```

Comment:
So how does this interact with the other channels? If I'm reading this code correctly then if I turn OOC on then toggle all on and off, then OOC will *look* down but still be up.

Reply:
Good point, I think it still works as intended iirc. There can be two enums of OOC existing in the list of "filtered channels". So if you un-toggled "ALL" it would actually still contain OOC and continue filtering until you un-toggle the OOC button itself. Ideally "ALL" should just flip button states then to solve this. Is there a method to do so with the current buttons?